server/controllers/recommendation/ppcs/indeterminateSentence/edit/editOffenceController.test.ts

```diff
@@ -0,0 +1,130 @@
+      ],
+    },
+  })
+  const updatedOffenceDescription = faker.lorem.sentence()
+  const updatedOffenceDescriptionComment = faker.lorem.sentence()
+
+  describe('get', () => {
+    const req = mockReq()
+    const res = mockRes({
+      locals: {
+        recommendation,
+      },
+    })
+    const next = mockNext()
+    beforeEach(async () => {
+      ;(ppudReferenceList as jest.Mock).mockResolvedValue({ values: ['one', 'two', 'three'] })
+      await editOffenceController.get(req, res, next)
+    })
+    describe('Sets response locals correctly:', () => {
+      describe('Page Data', () => {
+        it('- is defined', async () => {
+          expect(res.locals.existingOffenceDescription).toBeDefined()
+          expect(res.locals.allOffences).toBeDefined()
+        })
+      })
+    })
+    it('Does not set any error message', async () => expect(res.locals.errorMessage).toBeUndefined())
+    it('Calls render for the expected page', async () => expect(res.render).toHaveBeenCalledWith(expected.render.path))
+    it('Executes the next function ', async () => expect(next).toHaveBeenCalled())
+  })
+
+  describe('post', () => {
+    const recommendationId = faker.number.int().toString()
+    const basePath = `/recommendations/${recommendationId}/`
+    const next = mockNext()
+
+    const validReq = mockReq({
+      params: { recommendationId },
+      body: {
+        offenceDescription: updatedOffenceDescription,
+        offenceDescriptionComment: updatedOffenceDescriptionComment,
+      },
+    })
+    const res = mockRes({
+      locals: {
+        recommendation,
+        urlInfo: { basePath },
+      },
+    })
+
+    describe('Given a valid offence description is submitted', () => {
+      beforeEach(async () => {
+        await editOffenceController.post(validReq, res, next)
+      })
+      it('Then there are no errors', async () => expect(validReq.session.errors).toBeUndefined())
+      it('Then the recommendation is updated with the expected details', async () =>
+        expect(updateRecommendation).toHaveBeenCalledWith({
+          recommendationId,
+          valuesToSave: {
+            bookRecallToPpud: {
+              ...recommendation.bookRecallToPpud,
+              ppudIndeterminateSentenceData: {
+                ...recommendation.bookRecallToPpud.ppudIndeterminateSentenceData,
+                offenceDescription: updatedOffenceDescription,
+                offenceDescriptionComment: updatedOffenceDescriptionComment,
+              },
+            },
+          },
+          featureFlags: res.locals.flags,
+          token: res.locals.user.token,
+        }))
+      it('Then it redirects to the Sentence to Commit (Indeterminate page)', async () =>
+        expect(res.redirect).toHaveBeenCalledWith(303, `${basePath}sentence-to-commit-indeterminate`))
+      it('Does not execute the next function ', async () => expect(next).not.toHaveBeenCalled())
+    })
+    describe('Given no offence has been selected', () => {
```

Comment:
Is it possible to select an invalid option? Or is no option selected the only invalid case?